letsencrypt-win-simple/Plugin/SanByDomainPlugin.cs

```diff
@@ -0,0 +1,288 @@
+            }
+            return totalTarget;
+        }
+
+        //Copied from IISSiteServerPlugin
+        private static void ProcessTotaltarget(Target totalTarget, List<Target> runSites)
+        {
+            if (!Program.CentralSsl)
+            {
+                var pfxFilename = Program.GetCertificate(totalTarget);
+                X509Store store;
+                X509Certificate2 certificate;
+                Log.Information("Installing Non-Central SSL Certificate in the certificate store");
+                Program.InstallCertificate(totalTarget, pfxFilename, out store, out certificate);
+                if (Program.Options.Test && !Program.Options.Renew)
+                {
+                    Console.WriteLine($"\nDo you want to add/update the certificate to your server software? (Y/N) ");
+                    if (!Program.PromptYesNo())
+                        return;
+                }
+                Log.Information("Installing Non-Central SSL Certificate in server software");
+                foreach (var site in runSites)
+                {
+                    site.Plugin.Install(site, pfxFilename, store, certificate);
+                    if (!Program.Options.KeepExisting)
+                    {
+                        Program.UninstallCertificate(site.Host, out store, certificate);
+                    }
+                }
+            }
+            else if (!Program.Options.Renew || !Program.Options.KeepExisting)
+            {
+                var pfxFilename = Program.GetCertificate(totalTarget);
+                //If it is using centralized SSL, renewing, and replacing existing it needs to replace the existing binding.
+                Log.Information("Updating new Central SSL Certificate");
+                foreach (var site in runSites)
+                {
+                    site.Plugin.Install(site);
+                }
+            }
+
+            if (Program.Options.Test && !Program.Options.Renew)
+            {
+                Console.WriteLine(
+                    $"\nDo you want to automatically renew this certificate in {Program.RenewalPeriod} days? This will add a task scheduler task. (Y/N) ");
+                if (!Program.PromptYesNo())
+                    return;
+            }
+
+            if (!Program.Options.Renew)
+            {
+                Log.Information("Adding renewal for {binding}", totalTarget);
+                Program.ScheduleRenewal(totalTarget);
+            }
+        }
+
+        public string GetDomain(string input)
+        {
+            try
+            {
+                var nodes = input.Split('.');
+                if (nodes.Length > 2)
+                    return (nodes[nodes.Length - 2] + "." + nodes[nodes.Length - 1]).ToLowerInvariant().Trim();
+
+                return input.ToLowerInvariant().Trim();
+            }
+            catch
+            {
+                return string.Empty;
+            }
+        }
+    }
+
+    public class Domain
```

Comment:
This needs to be a class added to the solution, not a class added directly in your plugin.